client/extensions/woocommerce/app/products/product-form-variations-row.js

```diff
@@ -1,111 +1,231 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import Gridicon from 'gridicons';
 import { localize } from 'i18n-calypso';
+import { head } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import formattedVariationName from 'woocommerce/lib/formatted-variation-name';
+import Button from 'components/button';
 import FormCurrencyInput from 'components/forms/form-currency-input';
 import FormDimensionsInput from 'woocommerce/components/form-dimensions-input';
 import FormTextInput from 'components/forms/form-text-input';
 import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
+import ImagePreloader from 'components/image-preloader';
+import ProductImageUploader from 'woocommerce/components/product-image-uploader';
+import Spinner from 'components/spinner';
+
+class ProductFormVariationsRow extends Component {
+	static propTypes = {
+		product: PropTypes.object.isRequired,
+		variation: PropTypes.object.isRequired,
+		manageStock: PropTypes.bool,
+		onShowDialog: PropTypes.func,
+		editProductVariation: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+		const { variation } = props;
+		const image = variation && variation.image || {};
+
+		this.state = {
```

Comment:
`id` and `src` both eventually end up in Redux variations UI state after they have uploaded -- the rest of these are used for previews/progress as a image uploads so I think it's okay to leave in the component unless we find we need access to it from elsewhere.